graphqlapi/src/test/java/io/stargate/graphql/schema/schemafirst/processor/MappingModelTest.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright The Stargate Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.stargate.graphql.schema.schemafirst.processor;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.*;
+
+import graphql.schema.idl.SchemaParser;
+import graphql.schema.idl.TypeDefinitionRegistry;
+import io.stargate.db.schema.ImmutableKeyspace;
+import org.junit.jupiter.api.Test;
+
+class MappingModelTest {
+
+  @Test
+  public void shouldBuildAMappingModelWithOnePrimaryKey() {
+    TypeDefinitionRegistry typeDefinitionRegistry =
+        new SchemaParser()
+            .parse(
+                "type User { id: ID! name: String username: String } "
```

Comment:
my understanding is that we need to add a way to pass the clustering key in the entity and query, is that right?
As far as I can see the `!` syntax is for Primary keys. @olim7t what syntax should we use for adding other primary keys?

Reply:
`!` means non-null in GraphQL, it's not directly related to primary keys (although it makes sense to mark PK fields as such, since that's the case in Cassandra).

We already have syntax to add partition and clustering columns. The explicit way is via the `cql_column` directive:
```graphql
type Foo {
  pk1: Int! @cql_column(partitionKey: true)
  pk2: Int! @cql_column(partitionKey: true)
  cc1: Int! @cql_column(clusteringOrder: ASC)
  cc2: Int! @cql_column(clusteringOrder: DESC)
}
# produces CQL `PRIMARY KEY ((pk1,pk2), cc1, cc2)`
```
For convenience, we also have a convention that if the partition key is a single `ID`, `Uuid` or `TimeUuid` field, then you can omit the annotation if the field is the first in the object. That's what happens in your `User` type above:
```graphql
type User {
  id: ID! # implicit: @cql_column(partitionKey: true)
  name: String
  username: String
}
```
This would also work:
```graphql
type Foo {
  pk: Int!
  cc1: Int! @cql_column(clusteringOrder: ASC)
  cc2: Int! @cql_column(clusteringOrder: DESC)
}
# produces CQL `PRIMARY KEY (pk, cc1, cc2)`
```